ReaderApp.js

```diff
@@ -1532,8 +1582,11 @@ class ReaderApp extends React.PureComponent {
       // only want to show versionLanguage in results
       const removeLang = versionLanguage === "he" ? "en" : "he";
       data.result[removeLang] = "";
-      this.state.versionContents[pos] = data.result;
-      this.setState({versionContents: this.state.versionContents.slice(0)});
+      
+      const newVersionContents = [...this.state.versionContents];
```

Comment:
wrap this logic in setState callack and calculate `newVersionContents` with prevState